src/ripple/net/ShardDownloader.md

```diff
@@ -175,9 +175,9 @@ The `download_path` field of the `shard_db` entry is used to determine where to
 # https://ripple.com/build/history-sharding
 [shard_db]
 type=NuDB
-path=/var/lib/rippled/db/shards/nudb
-download_path=/var/lib/rippled/db/shards/
-max_size_gb=50
+main_storage_path=/var/lib/rippled/db/shards/nudb
+partial_shard_download_path=/var/lib/rippled/db/shards/
+max_historical_shards=50
```

Comment:
> I don't like some of these name changes:
> 
>     1. `path` to `main_storage_path` feels redundant. A path is clearly talking about storage, and if it doesn't have a prefix, it makes sense to assume that it's the "main" path. Additionally, some operators were already using shards, and now they have to make an arbitrary change that doesn't really add any value.
> 
>     2. `download_path` to `partial_shard_download_path` isn't as eggregious, but we're already in the `[shard_db]` section, so "shard" in the name is also redundant. "partial" also feels unnecessary, though I suppose you could make an argument that it's more descriptive since fully downloaded shards aren't left there, but that feels like a really unnecessary addition since we don't have a competing `complete_download_path`.
> 
> 
> For both `path` and `download_path`, since existing configs use the old values, and these name changes don't seem to add anything that can't be added with updated and clarified docs, I think these name changes should be rolled back
> 

I don't like renaming the two path keys either. It was suggested by @scottschurr on #3927 and I disagreed with the suggestion. See [this comment thread](https://github.com/ripple/rippled/pull/3297#discussion_r402649904). I'm fine with reverting the path names but I think we should arrive at a consensus to avoid further thrashing. Pinging @miguelportilla

Reply:
> I'm OK with switching from a size to a count, but I'm curious what the motivation for the change was. The reason it might be a problem is that it's pretty easy for an operator to look at their device like, "I have 100Gb free on this disk. I'll let shards have 60Gb of it." Whereas there's no real way to definitively say how much space 50 shards will take up, especially since they vary by size. Would it be worth adding an optional "minimum free space" field so that any download stops if the disk gets less than that amount of space?

The decision to control storage with a shard count rather than a byte count came from @miguelportilla. This was partly motivated by a desire to mimic the node store's approach of specifying ledger counts to limit db size, placing the burden of ensuring the availability of sufficient storage capacity on the user. Also note that the `DatabaseShardImp` does provide some help by issuing a warning if the available capacity is insufficient to store the full quantity of shards specified by the shard count. However, this warning is just meant to be supplementary and we expect the user to be responsible for providing ample storage for any server storing shard history.

Currently, the `DatabaseShardImp` already implements some notion of "minimum free space" by estimating the average size of a shard and minimally requiring that this much space must be available before storing an additional shard. Unfortunately this messy calculation is not ideal as shard sizes vary and are positively correlated with the shard index (later shards are typically larger in size). We have tentative plans to improve this somewhat by adding a sizeable buffer to this minimum size requirement (this change is partly inspired by the addition of `DeterministicShards`, as the capacity requirements during the finalization of shards may be double the size of the finalized shard). So in short, we're in the process of improving capacity management, but as of yet we have no plans of doing this with configuration properties.

Reply:
> I don't like some of these name changes:
> 
>     1. `path` to `main_storage_path` feels redundant. A path is clearly talking about storage, and if it doesn't have a prefix, it makes sense to assume that it's the "main" path. Additionally, some operators were already using shards, and now they have to make an arbitrary change that doesn't really add any value.
> 
>     2. `download_path` to `partial_shard_download_path` isn't as eggregious, but we're already in the `[shard_db]` section, so "shard" in the name is also redundant. "partial" also feels unnecessary, though I suppose you could make an argument that it's more descriptive since fully downloaded shards aren't left there, but that feels like a really unnecessary addition since we don't have a competing `complete_download_path`.
> 
> 
> For both `path` and `download_path`, since existing configs use the old values, and these name changes don't seem to add anything that can't be added with updated and clarified docs, I think these name changes should be rolled back
> 

Rolling back.